### ISSUE TITLE
DFlash speculative decoding crashes: GGML_ASSERT in encode (n_ubatch) and sampling (logits nullptr)

### ISSUE THREAD

**Issue by thadreber-web** [spiritbuun/buun-llama-cpp#21]
## Bug Report

DFlash speculative decoding crashes on the SD-076 branch (commit 6b17bcc) when serving inference requests with the Qwen3.6-35B-A3B target model and z-lab/Qwen3.6-35B-A3B-DFlash draft model.

Two distinct crashes occur:

### Crash 1: GGML_ASSERT in encode (n_ubatch)

When draft context is shared with target (-c 65536, no -cd flag):

```
/raid/models/llama-cpp-dflash/src/llama-context.cpp:2320: GGML_ASSERT(cparams.n_ubatch >= n_tokens && "encoder requires n_ubatch >= n_tokens") failed
#3 llama_context::encode(llama_batch const&)
#4 llama_decode()
```

The DFlash draft model uses non-causal (encoder-style) attention. When the draft model's n_ctx is set to 65536 (same as target), the encoder batch exceeds n_ubatch (512). This triggers the assertion in `encode()`.

### Crash 2: GGML_ASSERT in sampling (logits nullptr)

When draft context is limited via -cd 512, the server starts and processes the first prompt successfully, but crashes on the **second request** (short prompt, 24 tokens):

```
/raid/models/llama-cpp-dflash/common/sampling.cpp:152: GGML_ASSERT(logits != nullptr) failed
get_logits_ith: invalid logits id 2, reason: batch.logits[2] != true
#3 common_sampler_sample()
#4 common_sampler_sample_and_accept_n()
#5 server_context_impl::update_slots()
```

The draft model's forward pass does not produce logits for all tokens in the speculative batch. The `common_sampler_sample` function tries to sample from a position where logits were never computed (batch.logits[2] is false).

### Environment

- Hardware: NVIDIA GB10 (DGX Spark, aarch64)
- OS: Ubuntu 24.04 (aarch64)
- Build: Release, GGML_CUDA=ON, GGML_CUDA_FA=ON, GGML_CUDA_GRAPHS=ON, GGML_NATIVE=ON
- Branch: experiment/SD-076-moe-dflash-tape (commit 6b17bcc)

### Reproduction

**Target model:** samuelcardillo/Carnice-Qwen3.6-MoE-35B-A3B-GGUF (Q4_K_M)
**Draft model:** z-lab/Qwen3.6-35B-A3B-DFlash converted to GGUF F16 with Qwen3.6 tokenizer merges

```bash
# Crash 1 (n_ubatch assertion):
./llama-server -m target.gguf -md draft.F16.gguf -ngl 999 -ngld 999 \
  -ctk turbo4 -ctv turbo4 -c 65536 --flash-attn on --draft 8 --port 8081

# Crash 2 (logits nullptr) - fires on second inference request:
./llama-server -m target.gguf -md draft.F16.gguf -ngl 999 -ngld 999 \
  -ctk turbo4 -ctv turbo4 -c 65536 -cd 512 --flash-attn on --draft 8 --port 8081
```

### Additional Notes

- `the target and draft vocabs are not compatible - tokens will be translated between the two` is logged even though both models use the Qwen3.6 tokenizer (248320 vocab, 247587 merges). Possible vocab mismatch in special tokens or padding.
- Draft model metadata: 8 layers, 2048 embedding, non-causal attention, block_size=16, Gated Delta Net
- With -cd 512, draft compute buffer was ~1GB CUDA per slot (vs 3.6GB with -c 65536 shared)
- The first inference request (long prompt, 27k+ tokens) completes successfully before crash 2 occurs on the next request
